### ISSUE TITLE
Physical facet of computation - different from app development

### ISSUE THREAD

**Issue by sanjarcode** [sanjarcode/gui-apps#15]
> ❓career change decision.

Since college upto now, I have been working on software that interacts with humans, or at most, stays within a computer.

In other words, all my computation/software knowledge has almost no significant _physical_ effects, because no actuator/sensor is present, except for humans of course 😅.

Example: web (UI + backend) development is used by humans. Something like robotics is much more tangible and has more ability to make physical changes to the environment around us.

> In short, there are two types of computation systems:
1. Computation as an end goal, here product = computation + information.
2. Computation as a means to physical changes. Here product = computation + matter manipulation. See [CPS](https://en.wikipedia.org/wiki/Cyber-physical_system)

> I have explored 1

> I am interested in 2. Is it time to start learning about it?

> Note: Learning robotics has an intelligence part too, so learning fundamentals of AI (and so ML) will be crucial.

> Decision to make: how much of app development is OK before I stop diving deep into it, and start exploring this other branch of computation (computation with physical effects).

In short, what should be my plan?

> Note: there's a general "belief" that web development is a more in-demand skill than robotics. It's much more scalable than robotics, and robotics needs upfront capital.

However, none of these reasons are acceptable to "not do" robotics.

> What do you think about this, should I pursue this?

**Comment by sanjarcode** [comment]
> I'm not not bored, with making CRUD apps.
Apart from things like system design and code design (design patterns n all), I know programming well.

> That not not was intentional.

<img width="350px" alt="Screenshot_20220715-224146_Chrome.png" src="https://user-images.githubusercontent.com/48156230/179279071-e4dd48e0-5345-4db6-b4b0-a1859e69e8a2.png" />

<img width="350px" alt="Screenshot_20220715-224219_Chrome.png" src="https://user-images.githubusercontent.com/48156230/179279078-bfab1831-5325-4d1b-9776-bfd3bb1d4c66.png" />

> I'm definitely not bored, but the screenshots represent a small portion of how I feel...

<img width="350px" alt="Screenshot_20220715-224535_Chrome.png" src="https://user-images.githubusercontent.com/48156230/179279161-d0aa2630-5588-4c94-9ee1-cb3c6f271faf.png" />

<img width="350px" alt="Screenshot_20220715-224852_Chrome.png" src="https://user-images.githubusercontent.com/48156230/179279183-22b1fece-5c19-4177-808f-d3bbcaf7d792.png" />

> My current career path is simple: 1. Learn app development for the web: a. Ul b. Backend 2. Learn distributed system design 3. Mobile development (optional)
And in the process - also learn about programming - design patterns, paradigms.
I want to finish all this in < 2 years.

> When will I learn robotics, that is the question?
Should I learn them together -
i.e. app
development with robotics, or learn in order, or learn only one thing.

> **nextL** Reflecting on current career plan

**Comment by sanjarcode** [comment]
https://en.m.wikipedia.org/wiki/Center_for_Bits_and_Atoms

> MIT's Center for Bits and Atoms is an ambitious interdisciplinary initiative that is looking beyond the end of the Digital Revolution to ask how a functional description of a system can be embodied in, and abstracted from, a physical form. These simple, profound questions date back to the beginning of modern manufacturing and before that to the origins of natural science, but they have revolutionary new implications that follow from the recognition of the computational universality of physical systems. We can no longer afford to ignore nature's capabilities that have been neglected by conventional digital logic; it is at the boundary between the content of information and its physical representation that many of science's greatest technological, economic, and social opportunities and obstacles lie.

**Comment by sanjarcode** [comment]
[Automation, AI change happens first in the digital world, then in the physical world](https://youtu.be/WNYuUxeHYpE&t=52m10s), what does this mean?

Also, unlike how I feel people think, i.e. not considering digital and physical effects to software, Sam Altman does think about it. It'd be good to know ideas, or atleast data that I can use in this regard.
